sdk/core/azure-core/inc/azure/core/http/transport.hpp

```diff
@@ -66,6 +66,12 @@ namespace Azure { namespace Core { namespace Http {
      * @return A reference to this instance.
      */
     HttpTransport& operator=(const HttpTransport& other) = default;
+
+    /**
+     * @brief Returns true if the HttpTransport supports WebSockets (the ability to
+     * communicate bidirectionally on the TCP connection used by the HTTP transport).
+     */
+    virtual bool SupportsWebSockets() const { return false; }
```

Comment:
Wonder if we should have a guidline for functions and follow the `IsXYZ()` pattern.  For this method it would be `IsWebSocketsSupported()`

Reply:
A *really* good question. I slightly like the `'if (SupportsWebSockets())` syntax because it reads closer to english, however the `SupportsXxx` framing is semantically much more awkward than `IsXxxSupported`.